pkg/kubelet/cm/dra/manager.go

```diff
@@ -208,43 +197,54 @@ func (m *ManagerImpl) UnprepareResources(pod *v1.Pod) error {
 		}
 
 		// Skip calling NodeUnprepareResource if other pods are still referencing it
-		if len(claimInfo.podUIDs) > 1 {
+		if len(claimInfo.PodUIDs) > 1 {
+			// It OK not checkpoint the deletePodReference() here and below because if it
+			// missing after kubelet restart it means that we have never returned success from the
+			// UnprepareResources() call for this pod and it never deleted by the kubelet.
+			// Eventaully we capture all deletePodReference() calls at the end of the UnprepareResources().
```

Comment:
```suggestion
// We delay checkpointing of this change until this call returns successfully.
// It is OK to do this because we will only return successfully from this call if
// the checkpoint has succeeded. That means if the kubelet is ever restarted
// before this checkpoint succeeds, we will simply call into this (idempotent)
// function again.
```